feat(Firebase): agregar endpoints para mostrar y actualizar el token FCM

asoto-ideasdigitales opened the pull request:
Actualizacion de endpoints. (firebase config) 
FCM Token a usuario registrado

asoto-ideasdigitales committed:
feat(Firebase): agregar endpoints para mostrar y actualizar el token FCM, actualizacion de enpoint datos usuario

prodev443 reviewed (changes requested):
Hacen falta tests, favor de desarrollarlos

asoto-ideasdigitales commented:
> Hacen falta tests, favor de desarrollarlos
<img width="445" height="224" alt="image" src="https://github.com/user-attachments/assets/11f183bd-1587-4637-91e8-790186c9a288" />

Aprovechando se corrigen test. UserTest, RolesTest
Todos los tests. ok 
<img width="283" height="50" alt="image" src="https://github.com/user-attachments/assets/544301ed-f141-4e76-a6ef-a4952bb500c8" />

asoto-ideasdigitales committed:
feat(Roles): actualizar rutas de roles para consistencia en la API
feat(Firebase): mejorar pruebas de configuración de Firebase y permisos
fix(User): refactorizar prueba de permisos de búsqueda de usuarios

JavierAgueroCL commented on app/Http/Controllers/Api/FirebaseConfigController.php:
```diff
@@ -7,9 +7,68 @@
+            ], 404);
+        }
+
+        $raw = @file_get_contents($path);
```
El uso de @ para suprimir errores no es una buena práctica; mejor usar try-catch

asoto-ideasdigitales replied:
corregido

JavierAgueroCL commented on app/Http/Controllers/Api/FirebaseConfigController.php:
```diff
@@ -22,4 +81,20 @@ public function update(FirebaseConfigRequest $request): JsonResponse
         return response()->json(['message' => 'Firebase config saved'], 200);
     }
+
+    public function updateFcmToken(Request $request): JsonResponse
```
El método updateFcmToken debería estar en UserController por principio de responsabilidad única

JavierAgueroCL reviewed (changes requested):
.

asoto-ideasdigitales committed:
feat(Firebase): cambiar método para actualizar el token FCM a usercontroller  y manejar errores en la carga de configuración
